common/lib/xmodule/xmodule/modulestore/mongo/base.py

```diff
@@ -1010,13 +1010,29 @@ def extract_course_summary(course):
                 if field in course['metadata']
             }
 
-        course_org_filter = kwargs.get('org')
-        query = {'_id.category': 'course'}
+        course_records = []
+        query = {}
+        query['_id.category'] = 'course'
+        course_keys = kwargs.get('course_keys')
+
+        if course_keys:
+            for course_key in course_keys:
+                course_query = {
+                    '_id.{}'.format(value_attr): getattr(course_key, key_attr)
```

Comment:
@noraiz-anwar Can you checkout if we can query by `OR` operator? This way, we can use `.find` rather than quering db for every course record.

Reply:
https://docs.mongodb.com/manual/tutorial/query-documents/

Reply:
@awaisdar001 Yes, we can use `OR` operator but lets discuss cost of using it first.